### PR TITLE
middleware: add rootpassword RPC

### PR DESCRIPTION
**Pull request by 0xB10C**
This PR adds the RootPassword RPC. It can set the root password and enable/disable the root login.

_This PR builds on top of the work done in #163  (only adds 6439124a7d17935f4959f595b286096e9c9d3fb2). As soon as that one is merged into master I'll rebase this to master. I'll leave this marked as WIP in the meanwhile._

### CODE REVIEW

**Comment by 0xB10C** [comment]
Rebased on top of changes in #163

File: middleware/src/middleware.go [hunk 1]
```diff
@@ -251,6 +256,74 @@ func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessag
...
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.DisableRootLogin:
+		log.Println("Disabling the root login via the config script")
+		out, err := middleware.runBBBConfigScript("disable", "root_pwlogin", "")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
```
Comment (TheCharlatan) [inline middleware/src/middleware.go]: Can you split this method up into a SetRootPassword method and a RootLoginPolicy method?

**Comment by 0xB10C** [comment]
Will be split up into:

| RPC                | Input                 | Output                 |                  
|--------------------|-----------------------|-----------------------|
| EnableRootLogin    |`dummyArg bool`| ErrorResponse                                       |
| DisableRootLogin   | `dummyArg bool` | ErrorResponse                                       |
| SetRootPassword    | password              | ErrorResponse                               |

**Comment by 0xB10C** [comment]
What happend here?

> `0xB10C wants to merge 0 commits into digitalbitbox:master from 0xB10C:2019-08-middleware-set-rootpw-rpc`

**Pull request closed without merging**